webrender/src/prim_store.rs

```diff
@@ -1476,7 +1490,7 @@ impl PrimitiveStore {
             (local_rect, xf_rect.bounding_rect)
         };
 
-        if !self.update_clip_task(
+        if perform_culling && may_need_clip_mask && !self.update_clip_task(
```

Comment:
It doesn't appear obviously correct to me that we move the `perform_culling` check before the `update_clip_task`. This will result in stale `metadata.screen_rect` and `metadata.clip_task` being present in the rendered primitives.

Reply:
```perform_culling``` is effectively a constant - it never changes from initial value. Therefore, the clip task will never be set, and the screen rect should only depend on the previous culling values, not part of the clip stack.